lib/ancestry.rb

```diff
@@ -34,4 +34,75 @@ def self.default_update_strategy
   def self.default_update_strategy=(value)
     @@default_update_strategy = value
   end
+
+  # used for materialized path
+  class MaterializedPathString < ActiveRecord::Type::Value
+    def initialize(casting: :to_i, delimiter: '/')
+      @casting = casting&.to_proc
+      @delimiter = delimiter
+    end
+
+    def type
+      :materialized_path_string
+    end
+
+    # convert to database type
+    def serialize(value)
+      if value.kind_of?(Array)
+        value.map(&:to_s).join(@delimiter).presence
```

Comment:
not sure if we want an empty ancestry to have `[]` or `nil`

technically null means that the value is undetermined. But in our case, we want to say "none" which is "" and not nil.